presto-main/src/main/java/com/facebook/presto/operator/ArrayRowUnnester.java

```diff
@@ -0,0 +1,63 @@
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.ColumnarRow;
+import com.facebook.presto.spi.type.RowType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class ArrayRowUnnester
+    implements Unnester
+{
+    private final List<Type> fieldTypes;
+    private ColumnarRow block;
+    private int position;
+    private int positionCount;
+
+    public ArrayRowUnnester(Type elementType)
```

Comment:
I thought about that. But setBlock expects a RowBlock, so it would be really strange for the constructor to take an arbitrary list of types. There are pros and cons for both. I think keeping the constructor consistent with other unnesters and make it obvious that this should be only used for RowType is better.